proof/archive.go

```diff
@@ -280,6 +292,16 @@ func NewFileArchiver(dirName string) (*FileArchiver, error) {
 		return nil, fmt.Errorf("unable to create proof dir: %w", err)
 	}
 
+	// We need to make sure that all our proof files have the new naming
+	// scheme. If they don't, we'll rename them now. This might take quite a
+	// while since we need to read each proof file and parse it to extract
+	// the outpoint and then rename it.
+	err := migrateOldFileNames(proofPath)
```

Comment:
So for now, we'll need to keep this partial migration check in place until we can make some other more breaking change? 

Alternatively, we could also move it to a diff prefix sub-folder, leaving the old proof files in tact. New versions would then only update the new proof files. This might be safer than trying to mass rename all the proof files users have on disk today.

Reply:
I think renaming is fine... I did add the check that we only try to rename files that actually have the old format. That way nothing should be logged in subsequent runs and the detection should be really fast. And then in a couple of versions we can probably just remove this code. Worst case situation would be that the file based archive doesn't find the proofs anymore if the migration code wasn't run for a user that skips many versions when upgrading.

Reply:
SGTM, we can always restore directly from the DB if needed.